server/src/main/java/org/elasticsearch/index/engine/SegmentsStats.java

```diff
@@ -50,12 +54,11 @@ public SegmentsStats(StreamInput in) throws IOException {
         maxUnsafeAutoIdTimestamp = in.readLong();
 
         final int size = in.readVInt();
-        final ImmutableOpenMap.Builder<String, FileStats> files = ImmutableOpenMap.builder(size);
+        files = Maps.newMapWithExpectedSize(size);
         for (int i = 0; i < size; i++) {
             FileStats file = new FileStats(in);
             files.put(file.getExt(), file);
         }
-        this.files = files.build();
```

Comment:
Could be simplified with `this.files = in.readMapValues(FileStats::new, FileStats::getExt);`